api/app/models/comment.rb

```diff
@@ -28,9 +29,63 @@ class Comment < ApplicationRecord
                       counter_cache: :children_count
   has_many :children, class_name: "Comment", foreign_key: :parent_id, dependent: :destroy,
                       inverse_of: :parent, counter_cache: :children_count
-  has_many :flags, as: :flaggable, dependent: :destroy, inverse_of: :flaggable
+  has_one :comment_created_event, -> { where event_type: Event::COMMENT_CREATED },
+          class_name: Event,
+          as: :subject,
+          dependent: :destroy,
+          inverse_of: :subject
+  delegate :project, to: :subject
 
   # Validations
   validates :body, :subject, presence: true
 
+  # Callbacks
+  after_create :notify_parent_creator!, :notify_project_editors!
```

Comment:
Ensure that there are specs for both of these behaviors. We have assertions around account creation, for example, that assert that the mailer will be called when things happen. You could use that approach here.